src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/applicationinfo.cpp

```diff
@@ -206,19 +227,89 @@ APPLICATION_INFO::TryCreateHandler(
             return S_OK;
         }
     }
+
     return S_FALSE;
 }
 
 VOID
 APPLICATION_INFO::ShutDownApplication(const bool fServerInitiated)
 {
+    IAPPLICATION* app = nullptr;
+    {
+        SRWExclusiveLock lock(m_applicationLock);
```

Comment:
We're now releasing the lock, running stop, then acquiring the lock and setting the properties to null. Before we would do everything in the lock.

Can this cause issues where shutdown is called a second time while stop is in progress and the stop would be called twice?